### ISSUE TITLE
Security Exception Targeting Android 11 (SDK 30)

### ISSUE THREAD

**Issue by hrossi** [mParticle/mparticle-android-sdk#56]
I'm facing a problem trying to bump our targetSdkVersion to 30

Every time that I try to log an event I receive the following exception:
```
1597947735.942 7717-7865 E/mParticle: Error saving message to mParticle DB.
    java.lang.SecurityException: getDataNetworkTypeForSubscriber
        at android.os.Parcel.createExceptionOrNull(Parcel.java:2373)
        at android.os.Parcel.createException(Parcel.java:2357)
        at android.os.Parcel.readException(Parcel.java:2340)
        at android.os.Parcel.readException(Parcel.java:2282)
        at com.android.internal.telephony.ITelephony$Stub$Proxy.getNetworkTypeForSubscriber(ITelephony.java:8762)
        at android.telephony.TelephonyManager.getNetworkType(TelephonyManager.java:3021)
        at android.telephony.TelephonyManager.getNetworkType(TelephonyManager.java:2985)
        at com.mparticle.internal.u.b(SourceFile:209)
        at com.mparticle.internal.t.a(SourceFile:93)
        at com.mparticle.internal.d.handleMessage(SourceFile:52)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.os.HandlerThread.run(HandlerThread.java:67)
```
I'm afraid that this exception is related to the following Android 11 behavior change [LINK](https://developer.android.com/preview/behavior-changes-11#connectivity)

Apps that use the [Telephony](https://developer.android.com/reference/android/provider/Telephony) will receive a security exception if they don't have the [Manifest.permission.WRITE_APN_SETTINGS](https://developer.android.com/reference/android/Manifest.permission#WRITE_APN_SETTINGS) permission (system permission).

**Comment by samdozor** [comment]
hey @hrossi thanks for the report. It seems the Android docs are a bit misleading here. The doc you linked states that `WRITE_APN_SETTINGS` is required to access the "telephony APN database", whereas [the docs](https://developer.android.com/reference/android/telephony/TelephonyManager#getNetworkType()) for [the `getNetworkType ` method we're calling](https://github.com/mParticle/mparticle-android-sdk/blob/master/android-core/src/main/java/com/mparticle/internal/MessageManager.java#L209)  state that only the more benign `READ_PHONE_STATE` is required.

Either way, given this behavior change we should be verifying that we have access prior to making the call. We will make that change ASAP for next week's SDK release.

**Comment by hrossi** [comment]
Hi @samdozor 

I appreciate your attention and quick response.

Please let me know if you need any further information.

**Comment by samdozor** [comment]
hi @hrossi this should be resolved in SDK 5.14.11 which is rolling out to Maven Central now.

**Comment by hrossi** [comment]
Hi @samdozor,

Thanks. I'll check it and I let you know if I face any other problem.
